Dockerfile

```diff
@@ -176,8 +176,22 @@ COPY vendor /go/src/github.com/docker/docker/vendor
 RUN go install -v github.com/cpuguy83/go-md2man \
 	github.com/BurntSushi/toml/cmd/tomlv
 
+# Build the tool for embedding resources in Windows binaries
+ENV RSRC_COMMIT e48dbf1b7fc464a9e85fcec450dddf80816b76e0
+RUN set -x \
+    && git clone https://github.com/akavel/rsrc.git /go/src/github.com/akavel/rsrc \
+    && (cd /go/src/github.com/akavel/rsrc \
+          && git checkout -q $RSRC_COMMIT \
+          && go build -v -o /go/bin/rsrc)
+
 # Wrap all commands in the "docker-in-docker" script to allow nested containers
 ENTRYPOINT ["hack/dind"]
 
 # Upload docker source
 COPY . /go/src/github.com/docker/docker
+
+# Compile the Windows resources into the sources
+RUN rsrc \
+       -manifest /go/src/github.com/docker/docker/hack/make/resources/docker.exe.manifest \
+       -ico /go/src/github.com/docker/docker/hack/make/resources/docker.ico \
+       -o   /go/src/github.com/docker/docker/pkg/winresources/rsrc.syso
```

Comment:
One issue here is that `pkg/...` is explicitly Go packages that are useful _outside_ the context of Docker, where this part isn't.

The generation also still needs to happen outside the `Dockerfile`, because even if the eventual plan is to build the official release daemon inside the `Dockerfile`, it's _supported_ to do so outside (especially since package managers usually need to be able to do that kind of thing).  It's fine that we install the tool in the `Dockerfile` (just like `go-md2man`), but it needs to be run from one of the scripts in `hack/make*` (and from what it appears to do, probably `hack/make/.go-autogen` -- https://github.com/docker/docker/blob/15b45113eb4aaeca9c289c22f772deead0024560/hack/make/.go-autogen, but that depends on why `resources.go` absolutely must pre-exist).